staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -18,19 +18,83 @@ package workqueue
 
 import (
 	"container/heap"
+	"fmt"
+	"strconv"
+	"strings"
 	"sync"
 	"time"
 
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/utils/clock"
 )
 
+// queueLogger is any type that provides the Log and Logf methods.
+type queueLogger interface {
```

Comment:
I chose a fairly small interface for this because I only think that the logging use desirable in debugging in test suites (as I have already needed to use it for).  It is convenient to be able to pass the test suite in as the logger here so that the queue activity is tabulated with any logging or output activity produced by the tests.  I think https://github.com/go-logr/logr might be worth doing if we expect to want to enable logging in non-test scenarios, but I don't think this would be desirable due to the performance implications.

Reply:
OK, we could keep this interface only if used for our debugging purposes only, but let's revisit this if we decide to expose the logging funcionality as part of an API: https://github.com/kubernetes/kubernetes/pull/112328#discussion_r1116958088